submissions/SofiiaTrokhymchuk/tiny-js-oop/index.js

```diff
@@ -0,0 +1,83 @@
+class Inhabitant{
+   constructor(species, name, gender, saying){
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.properties = ['species', 'name', 'gender', 'saying'];
+   }
+
+   addFriends(friendsList){
+      this.friends = friendsList;
+      this.properties.push('friends');
+   }
+
+   toString(){
+      return this.properties.map(property => {
+         return (Array.isArray(this[property])) ?
+            (`<strong>${property}</strong>: ${this[property].join(", ")}`) :
+            (`<strong>${property}</strong>: ${this[property]}`)
+      }).join(", ");
+   }
+}
+
+class Animal extends Inhabitant{
+   constructor(species, name, gender, saying){
+      super(species, name, gender, saying);
+      this.legs = 4;
+      this.properties.push('legs');
```

Comment:
Every class should handle the props it **owns** immediately.

> A parent class shouldn't know anything about its children's props. Child classes must have good reasons to assign or refer to any parent's props. There are no such reasons in this task. Method overloading is to the rescue when building an inhabitant's presentation string.

Method overloading and explicit call to the inherited version of a method are to the rescue.
Task and original project repo offer links to the materials that you will find helpful in these regards.